### PR TITLE
Simplification on boot modes logic. Now any mode can be triggered programmatically or scheduled for the next boot.

### PR DESCRIPTION
**Pull request by flaviostutz**
The modes are the same: STANDALONE, CONFIG AND NORMAL

 - The application can put Homie in any one of the modes using Homie.setBootMode(...)
 - The application can schedule Homie to boot once in a determined mode using Homie.setBootModeNextBoot(...)
 - Overall refactoring on BootMode, removal of unused code and small fixes

Why this was implemented:
 - In my case, I want to enter config mode automatically when I detect that Wifi or MQTT are not working well in the next time the user turns on my device. Using this patch, I can call setBootModeNextBoot(...)
 - During Config mode, when there is a valid previous configuration, I will reboot the device automatically back to NORMAL mode after a timeout period (3min and so)
 - In my case, I don't have any buttons on the device, that's why I have to create "smart" ways to enter and exit config mode
 - All those new/refactored functions were teste by the App.

If you accept this PR, I will update the documentation. What do you think?

### CODE REVIEW

File: src/Homie/Config.cpp [hunk 1]
```diff
@@ -262,8 +281,8 @@ bool Config::patch(const char* patch) {
     return true;
 }
 
-BootMode Config::getBootMode() const {
-  return _bootMode;
+bool Config::isValid() {
```
Comment (euphi) [inline src/Homie/Config.cpp]: bool Config::isValid() **const**

File: src/Homie/Config.cpp [hunk 2]
```diff
@@ -168,26 +167,46 @@ void Config::erase() {
   if (!_spiffsBegin()) { return; }
 
   SPIFFS.remove(CONFIG_FILE_PATH);
-  SPIFFS.remove(CONFIG_BYPASS_STANDALONE_FILE_PATH);
+  SPIFFS.remove(CONFIG_NEXT_BOOT_MODE_FILE_PATH);
 }
 
-void Config::bypassStandalone() {
+void Config::setBootModeOnNextBoot(BootMode bootMode) {
```
Comment (euphi) [inline src/Homie/Config.cpp]: This does not change the `Config` instance, so it can be defined as `const`.

File: src/Homie/Config.cpp [hunk 3]
```diff
@@ -168,26 +167,46 @@ void Config::erase() {
...
+    bootModeFile.close();
+    Interface::get().getLogger().printf("Setting next boot mode to %d\n", bootMode);
+  }
 }
 
-bool Config::canBypassStandalone() {
-  if (!_spiffsBegin()) { return false; }
-
-  return SPIFFS.exists(CONFIG_BYPASS_STANDALONE_FILE_PATH);
+BootMode Config::getBootModeOnNextBoot() {
```
Comment (euphi) [inline src/Homie/Config.cpp]: This method should be `const`.

**Comment by flaviostutz** [comment]
@marvinroger, could you please see if this is OK? I have another pull request to be submitted, but I want to know if this is OK...

File: src/Homie.hpp [hunk 4]
```diff
@@ -37,6 +37,11 @@ class HomieClass {
   void __setFirmware(const char* name, const char* version);
   void __setBrand(const char* brand) const;
 
+  static const BootMode MODE_UNDEFINED = BootMode::UNDEFINED;
```
Comment (marvinroger) [inline src/Homie.hpp]: For it to be consistent, can you please make the `BootMode` a `HomieBootMode` and include it the same way it works for https://github.com/marvinroger/homie-esp8266/blob/develop/src/HomieEvent.hpp? This way, the user could use `HomieBootMode::NORMAL` from his code.

**Comment by flaviostutz** [comment]
@marvinroger, Please check if this is ok now.

**Comment by flaviostutz** [comment]
I could not get my new commits to be included here. Closing and opening another pull request.

**Pull request closed without merging**